daisi/src/cpps/logical/amr/amr_logical_agent.cpp

```diff
@@ -0,0 +1,230 @@
+  daisi::cpps::CppsTCPMessage message;
+  message.addMessage({payload, 0});
+
+  auto packet = ns3::Create<ns3::Packet>();
+  packet->AddHeader(message);
+  socket_to_physical_->Send(packet);
+}
+
+void AmrLogicalAgent::sendTopologyToPhysical() {
+  auto topology_msg = amr::serialize(topology_);
+  sendToPhysical(topology_msg);
+}
+
+void AmrLogicalAgent::sendTaskToPhysical() {
+  material_flow::Task task = order_management_->getCurrentTask();
+
+  daisi::cpps::mrta::model::Ability ability(
+      42.0, daisi::cpps::mrta::model::LoadCarrier::kNoLoadCarrierType);
+
+  AmrOrderInfo amr_order_info(materialFlowToFunctionalities(task.getOrders(), current_position_),
+                              ability);
+  auto order_msg = amr::serialize(amr_order_info);
+
+  sendToPhysical(order_msg);
+}
+
+bool AmrLogicalAgent::connectionRequest(ns3::Ptr<ns3::Socket> socket, const ns3::Address &addr) {
+  // Accept all requests
+  return true;
+}
+
+void AmrLogicalAgent::newConnectionCreated(ns3::Ptr<ns3::Socket> socket, const ns3::Address &addr) {
+  socket_of_physical_ = socket;
+  socket_of_physical_->SetRecvCallback(
+      MakeCallback(&AmrLogicalAgent::readFromPhysicalSocket, this));
+
+  logAmrInfos();
+}
+
+void AmrLogicalAgent::checkSendingNextTask() {
+  // TODO check depending on AmrState and OrderState whether we are still busy or not
+
+  sendTaskToPhysical();
+}
+
+void AmrLogicalAgent::logAmrInfos() {
+  daisi::cpps::AGVLoggingInfo info;  // TODO rename to AMR
+
+  // general
+  info.friendly_name = "TODO";
+  info.serial_number = description_.getSerialNumber();
+  info.manufacturer = description_.getProperties().getManufacturer();
+  info.model_name = description_.getProperties().getModelName();
+  info.model_number = description_.getProperties().getModelNumber();
+
+  // kinematics
+  info.max_velocity = description_.getKinematics().getMaxVelocity();
+  info.min_velocity = description_.getKinematics().getMinVelocity();
+  info.max_acceleration = description_.getKinematics().getMaxAcceleration();
+  info.min_acceleration = description_.getKinematics().getMaxDeceleration();
+
+  // load handling unit
+  info.load_time = description_.getLoadHandling().getLoadTime();
+  info.unload_time = description_.getLoadHandling().getUnloadTime();
+  info.load_carrier_type =
+      daisi::cpps::mrta::model::LoadCarrier::kNoLoadCarrierType;  // TODO refactor
+  info.max_weight = 42;                                           // TODO refactor
+
+  // network
+
+  // retrieve physical info
+  ns3::Address physical_address;
+  socket_of_physical_->GetSockName(physical_address);
+
+  ns3::InetSocketAddress i_physical_address = ns3::InetSocketAddress::ConvertFrom(physical_address);
+  std::string physical_asset_ip = daisi::getIpv4AddressString(i_physical_address.GetIpv4());
+  uint16_t physical_asset_port = i_physical_address.GetPort();
+
+  // retrieve logical info
+  ns3::Address logical_address;
+  socket_of_physical_->GetSockName(logical_address);
```

Comment:
Maybe a typo. To get the address of the logical, this should be the listening socket.